src/main/java/net/javadiscord/javabot/systems/qotw/submissions/SubmissionManager.java

```diff
@@ -91,68 +104,116 @@ public WebhookMessageCreateAction<?> handleSubmission(@NotNull ButtonInteraction
 		);
 		log.info("Opened new Submission Thread for User {}", member.getUser().getAsTag());
 		return Responses.success(event.getHook(), "Submission Thread created", "Successfully created a new private Thread for your submission.");
+	}
 
+	public List<QOTWSubmission> getActiveSubmissions() {
+		return config.getSubmissionChannel().getThreadChannels()
+				.stream()
+				.map(this::getOrRetrieveSubmission).toList();
 	}
 
 	/**
 	 * Handles the "Delete Submission" Button.
 	 *
 	 * @param event The {@link ButtonInteractionEvent} that is fired upon use.
 	 */
-	public void handleThreadDeletion(ButtonInteractionEvent event) {
-		ThreadChannel thread = (ThreadChannel) event.getGuildChannel();
-		asyncPool.execute(()->{
-			try {
-				Optional<QOTWSubmission> submissionOptional = qotwSubmissionRepository.getSubmissionByThreadId(thread.getIdLong());
-				if (submissionOptional.isPresent()) {
-					QOTWSubmission submission = submissionOptional.get();
-					if (submission.getAuthorId() != event.getMember().getIdLong()) {
-						return;
+	public void handleThreadDeletion(@NotNull ButtonInteractionEvent event) {
+		config.getSubmissionChannel().getThreadChannels()
+				.stream().filter(t -> t.getIdLong() == event.getChannel().getIdLong())
```

Comment:
Why not just checking whether `event.getChannel()` is a thread and using `getType()` and converting it to a `ThreadChannel`?

Reply:
did that